src/data_access/SpotifyDataAccessObject.java

```diff
@@ -191,4 +195,171 @@ public void addSongsToPlaylist(String playlistId, List<String> songIds) throws I
     }
 
+        @Override
+        public List<String> getTopArtists(String searchQuery, int limit) throws IOException, InterruptedException, ParseException {
+            // Encode the search query to handle spaces and special characters
+            String encodedSearchQuery = URLEncoder.encode(searchQuery, StandardCharsets.UTF_8);
+
+            String url = "https://api.spotify.com/v1/search?q=" + encodedSearchQuery + "&type=artist&limit=" + limit;
+            String accessToken = SpotifyAuth.getAccessToken();
+
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .header("Authorization", "Bearer " + accessToken)
+                    .build();
+
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            JSONObject responseObject = (JSONObject) new JSONParser().parse(response.body());
+            JSONObject artistsObject = (JSONObject) responseObject.get("artists");
+            JSONArray items = (JSONArray) artistsObject.get("items");
+
+            List<String> artists = new ArrayList<>();
+            for (Object item : items) {
+                JSONObject artist = (JSONObject) item;
+                artists.add((String) artist.get("name"));
+            }
+            return artists;
+        }
+
+    public List<String> getArtistsTopTracks(List<String> queries, int limit) {
+        List<String> topTracks = new ArrayList<>();
+
+        for (String artistName : queries) {
+            try {
+                // Encode the artist name to handle spaces and special characters
+                String encodedArtistName = URLEncoder.encode(artistName, StandardCharsets.UTF_8);
+
+                String url = "https://api.spotify.com/v1/search?q=" + encodedArtistName + "&type=track&limit=" + limit;
+                String accessToken = SpotifyAuth.getAccessToken();
+
+                HttpClient client = HttpClient.newHttpClient();
+                HttpRequest request = HttpRequest.newBuilder()
+                        .uri(URI.create(url))
+                        .header("Authorization", "Bearer " + accessToken)
+                        .build();
+
+                HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+                JSONObject responseObject = (JSONObject) new JSONParser().parse(response.body());
+                JSONObject tracksObject = (JSONObject) responseObject.get("tracks");
+                JSONArray items = (JSONArray) tracksObject.get("items");
+
+                for (Object item : items) {
+                    JSONObject track = (JSONObject) item;
+                    String trackName = (String) track.get("id");
+                    topTracks.add(trackName);
+                }
+            } catch (IOException | InterruptedException | ParseException e) {
```

Comment:
nice, we should replace this catch condition instead of Exception for the other files